config/checker.go

```diff
@@ -58,28 +76,27 @@ func checkSlice(isReload bool, typeToCheck reflect.StructField, sliceToCheck ref
 	}
 
 	if sliceToCheck.Len() == 0 {
-		return fmt.Errorf("non optional slice field %s had zero length", typeToCheck.Name)
+		return []error{fmt.Errorf("non optional slice field %s in %s had zero length", typeToCheck.Name, parent)}
 	}
 
-	var err error
+	err := make([]error, 0)
```

Comment:
I would suggest using `errors` as the name for this slice, so that it's coherent with the rest of the functions and it helps understanding that it's not just a single error